libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -784,12 +838,18 @@ public static ExpressionEvaluator TimeTransform(string type, Func<DateTime, int,
                     string error = null;
                     IReadOnlyList<object> args;
                     (args, error) = EvaluateChildren(expr, state, options);
+                    var locale = DetermineLocaleInfo(state);
+                    var formatString = DefaultDateTimeFormat;
+                    if (error == null)
+                    {
+                        (formatString, locale, error) = DetermineFormatAndLocale(args, formatString, locale, 4);
+                    }
+
                     if (error == null)
                     {
                         if (args[1].IsInteger())
                         {
-                            var formatString = (args.Count() == 3 && args[2] is string string1) ? string1 : DefaultDateTimeFormat;
-                            (value, error) = NormalizeToDateTime(args[0], dt => function(dt, Convert.ToInt32(args[1])).ToString(formatString));
+                            (value, error) = NormalizeToDateTime(args[0], dt => function(dt, Convert.ToInt32(args[1])).ToString(formatString, locale));
```

Comment:
>locale [](start = 142, length = 6)

Is null allowed here?

Reply:
I don't think the locale here will be null due to DeterminLocale and DetermineFormatAndLocale method will take the CultureInfo.InvariantCulture as default setting.
After set the return locale value in DeterminFormatAndLocale to null. The ToString() method still works. It will return a result as the same as the Thread.CurrentThread.CurrentCulture.